### PR TITLE
Massive Scope Removal/Rework pr

### PR DESCRIPTION
**Pull request by Wisemonster**
## About The Pull Request
Combines #16030, #16011, #16005, #16003 (modified to scale with pop), and #15998
## Why It's Good For The Game
See the prs for reason, but to TLDR, reducing scopes range and availability is meant to help reduce the amount of off-screen kills, due to how un-fun they are to interact with on both sides.
## Changelog
:cl:
balance: Removed night vision traits from the autosniper and anti-material rifle
balance: Nerf MG-27, HMG, and HSG's scope to slightly less than mini scope
balance: nerf TAT scope on ATR-22 mounted flak gun, AT-36 anti tank gun, and AGLS-37 Kauser automatic grenade launcher to slightly less than mini scope
balance: Limits the miniscope attachments, SR-127, and mosin to 8 per alive human.
del: Replaced scopes with more range than a mini-scope on most weapons with a miniscope (exceptions are mounted weapons, anti-material rifle, and autosniper)
/:cl:

### CODE REVIEW

**Comment by Wisemonster** [comment]
It booted up for me, but this did involve a lot of pulling from other peoples prs, so let me know if I missed one or did one wrong.

**Comment by Woker2004** [comment]
friendly reminder to all the players that rework is another term for removal but with this aside, time for real critism.

Scopes, they are useful because each one has its gimmick, red dots allow for simply firing faster when aim mode, which is only on screen, mini rail scope only pushes roughly 3ish tiles off screen, but no fire rate boost, which if you are going to see off screen as as marine, you need lighting provided from Flares, CAS flares, fire, module lights, you get the idea, this also applies to the T-47 scope, you can't use the scope if you can't see.

Now the limitation thing. Limiting attachments just start causing more and more attachments to stand out, majority of people use mag-harness as is, this isn't something that will be really impactful except for the fact that we are going back down the line of soft removing equipment. scopes really aren't a problem with damage drop-off.

What could be done? Keep the scopes in, red dots can have its gimmick staying, rail mini scope could still do the 3ish far range, but potentially slows firerate slightly (like 0.9x), the T-47 could stay same range but slows down fire rate but 0.8x instead.

Removal is just not a good idea, nor is soft locking something that still is very niche, limiting it feels counter intuitive for what TGMC is about, customization, what this does is start changing TGMC back into CM which is just unformal which is what alot of people don't enjoy, which is why they play TGMC.

**Review by Kuro020 — APPROVED** [review]

**Comment by Barnet2** [comment]
Though I would have done some things differently, I like the overall idea. Will be interesting to see alongside the boiler tweak.

**Comment by gpinrgrmoma** [comment]
> red

this doesnt touch red dots at all

**Comment by SimplySmiley123** [comment]
> > red
> 
> this doesnt touch red dots at all

I don't think it was about red dots being in the PR or anything but mentioning them to make a point about the 2 other scopes.

Personally, this just fucks over regular sniper rifles even more, they'll just suck major balls now since their only reason to exist was to be a minor annoyance to the off-screen boiler (especialy SR-127, that thing will NOT work at miniscope range all that well, if at all), while the emplacements, do need their scopes for their IFF, without which they just become FF griefing tools let's be honest.

**Comment by chizzy376** [comment]
I am disappointed.

File: code/datums/gamemodes/infestation.dm [hunk 1]
```diff
@@ -33,6 +33,21 @@
 	for(var/i in GLOB.xeno_jelly_pod_turfs)
 		new /obj/structure/xeno/resin_jelly_pod(i, XENO_HIVE_NORMAL)
 
+	scale_weapons()
+
+/datum/game_mode/infestation/proc/scale_weapons(pop_override)
```
Comment (Lumipharon) [inline code/datums/gamemodes/infestation.dm:38]: autodoc. Although is there a reason this proc exists?

File: code/datums/gamemodes/infestation.dm [hunk 2]
```diff
@@ -33,6 +33,21 @@
...
+	if(!isnum(pop_override))
+		pop_override = length(GLOB.alive_human_list)
```
Comment (Lumipharon) [inline code/datums/gamemodes/infestation.dm:40]: this proc is run once on post setup... so doesn't include the majority of players that latejoin.

File: code/datums/gamemodes/infestation.dm [hunk 3]
```diff
@@ -33,6 +33,21 @@
...
+	var/item_ratio = floor(pop_override / INFESTATION_MARINE_PER_SCOPE)
+	for(var/obj/item/attachable/scope/mini AS in GLOB.vending_records["Attachments"])
+		if(mini == -1)
+			continue
+		mini = item_ratio
+	for(var/obj/item/weapon/gun AS in GLOB.vending_records["Marskman"])
+		if(gun == -1)
+			continue
+		gun = item_ratio
```
Comment (Lumipharon) [inline code/datums/gamemodes/infestation.dm:49]: uh, has this been tested?

File: code/modules/projectiles/guns/energy.dm [hunk 4]
```diff
@@ -765,7 +763,7 @@
 		/obj/item/attachable/bayonetknife,
 		/obj/item/attachable/bayonetknife/som,
 		/obj/item/attachable/magnetic_harness,
-		/obj/item/attachable/scope/unremovable/laser_sniper_scope,
+		/obj/item/attachable/scope/mini,
```
Comment (Lumipharon) [inline code/modules/projectiles/guns/energy.dm:766]: I am almost certain this hasn't been tested. laser sniper scope is hard built into the sprite.

File: code/modules/projectiles/guns/energy.dm [hunk 5]
```diff
@@ -780,7 +778,6 @@
 
 	gun_features_flags = GUN_CAN_POINTBLANK|GUN_ENERGY|GUN_AMMO_COUNTER|GUN_NO_PITCH_SHIFT_NEAR_EMPTY|GUN_AMMO_COUNT_BY_SHOTS_REMAINING
 	attachable_offset = list("muzzle_x" = 41, "muzzle_y" = 18,"rail_x" = 19, "rail_y" = 19, "under_x" = 28, "under_y" = 8, "stock_x" = 22, "stock_y" = 12)
-	starting_attachment_types = list(/obj/item/attachable/scope/unremovable/laser_sniper_scope)
```
Comment (Lumipharon) [inline code/modules/projectiles/guns/energy.dm:783]: as above.

File: code/modules/projectiles/guns/rifles.dm [hunk 6]
```diff
@@ -1499,7 +1490,7 @@
 		/obj/item/ammo_magazine/rifle/chamberedrifle/flak,
 	)
 	attachable_allowed = list(
-		/obj/item/attachable/scope/unremovable/tl127,
+		/obj/item/attachable/scope/mini,
```
Comment (Lumipharon) [inline code/modules/projectiles/guns/rifles.dm:1493]: you're removing all these unremoveable scope types with just the generic miniscope sprite?

File: code/modules/projectiles/guns/rifles.dm [hunk 7]
```diff
@@ -1522,10 +1513,7 @@
 	actions_types = list(/datum/action/item_action/aim_mode)
 	aim_fire_delay = 1 SECONDS
 
-	starting_attachment_types = list(
-		/obj/item/attachable/scope/unremovable/tl127,
-		/obj/item/attachable/stock/tl127stock,
-	)
+	starting_attachment_types = list(/obj/item/attachable/stock/tl127stock)
```
Comment (Lumipharon) [inline code/modules/projectiles/guns/rifles.dm:1516]: as above

File: code/modules/projectiles/guns/rifles.dm [hunk 8]
```diff
@@ -1561,7 +1549,7 @@
 	allowed_ammo_types = list(/obj/item/ammo_magazine/rifle/autosniper)
 	attachable_allowed = list(
 		/obj/item/attachable/autosniperbarrel,
-		/obj/item/attachable/scope/nightvision,
+		/obj/item/attachable/scope/marine,
```
Comment (Lumipharon) [inline code/modules/projectiles/guns/rifles.dm:1552]: as above, like none of these sprite changes have been tested, they're all going to look weird.

**Review by Lumipharon — CHANGES_REQUESTED** [review]

File: code/datums/gamemodes/infestation.dm (hunk 3 above)
Comment (Wisemonster) [reply code/datums/gamemodes/infestation.dm:49]: Didn't know of a way to test it on local, due to nuke war requiring a minimum number of xeno players to run

File: code/datums/gamemodes/infestation.dm (hunk 1 above)
Comment (Wisemonster) [reply code/datums/gamemodes/infestation.dm:38]: I was suggested to have the sniper weapons scale with pop, though this wasn't a maint request. If it's better to just hard limit the numbers, I can do that instead

**Comment by Wisemonster** [comment]
Alright, changed the guns with unremovable scopes back, but made them mini instead, so their sprites should be fine, and they keep their mini functionality

File: code/datums/gamemodes/infestation.dm (hunk 1 above)
Comment (Lumipharon) [reply code/datums/gamemodes/infestation.dm:38]: no I mean, does this have a reason to exist as a separate proc? Its only used in one place.

File: code/datums/gamemodes/infestation.dm (hunk 3 above)
Comment (Lumipharon) [reply code/datums/gamemodes/infestation.dm:49]: you know you can just press the start button a 2nd time and it lets you bypass those restrictions, right?

**Comment by CVSthePharmacy** [comment]
> Alright, changed the guns with unremovable scopes back, but made them mini instead, so their sprites should be fine, and they keep their mini functionality

what the fuck is wrong with you

**Comment by FosterGrime-Heymynameslouie** [comment]
image
(with regards to the mg scopes)

**Comment by TChalkman** [comment]
Gonna share my opinion on this in parts


Req weapons, specifically autosniper and AMR, DO need a nerf. Players are able to stack these and OBLITERATE anything on sight. That would need some tweaking to avoid having a firing line of hard hitting, unreachable gunners.

Stock snipers like the SR-127 are pretty niche and i don't see as often, and are substantially affected by their low fire-rate since they need to aim 90% of the time. The only thing getting killed by two snipers is like.... a stunlocked runner. They merely exist to deal damage through units and let chasers finish the job. 
Ive seen different changes for these...I dare say that limiting the amount of them on vendors would be inconsequential. But nerfing its range to basically a mounted gun can seriously kill these guns.


And now for the mounted/heavy ones like AT36, AGLs... These are weapons that require cover/support and good positioning to be used effectively, and can be easily shut down if flankers exist/marine is distracted. Plus, you can break the guns or just stun the gunner with any CC ability, and they've got to be insanely careful with their friendly fire. 
By nerfing their scopes into less than a miniscope.... you are turning these guns into abysmal dogshit. 

An AT36 already has to deal with stuff like: _constant calls with RO, dragging the gun around, carrying the ammo, making sure you follow ungaball to not get flanked, careful where you place yourself to not get instakilled by a crusher, friendly fire, marines bumping your gun all over the place on narrow areas, walls, defilers, "needing a friend" which means loading times for everything that isn't shooting the damn thing..._
These huge guns don't get so much use in the first place, and its only saving grace is their range. By making the range ALSO really small, you'd need to park the gun at the very frontline (where you can be easily flung off/killed), or you'll risk blowing up all marines.
In the case of machineguns, you'd also run on the same issue. You will need to expose yourself just to get in range, and can be easily taken down that way. You become a sitting duck AND someone who needs to be right infront of a xeno to shoot it. You are painting a fat red dot on your forehead if you play mounted guns then. 

How are marines supposed to deal with this? Carry deployable cades around, effectively doubling their setup time? Only for xenos/marines to push and having to move AGAIN? Babysitter engineer for every single mounted gunner?
They are indeed supposed to be vulnerable at close range... but we are making the line between long and short range so INSANELY THIN they almost become xeno bait.

At the very LEAST do not put these changes for heavy guns like the ATR, AT36, FK88.... these are weapons that already suffer enough and people barely play when theres other weapons that do their job WAY better like railguns or AMRs. Like I said in the scope mounted guns PR: If you want to kill all these guns this hard, just delete them

**Comment by maikilangiolo** [comment]
Already mentioned in the other PR. Mounted guns need their scopes. The miniscope on the machineguns that aren't the T27 is needed because they're borderline unusable without it. They're very slow to move around and deploy, and very, very vulnerable: easy to target the gunner, easy to break the gun. The larger cannons cannot be used without a scope, period. They have gigantic AOEs and are ever slower, the only thing this would do is cause them to fire inside the marines and grief with FF.

You want to reduce the accessibility of scopes? Absolutely fine from guns, but mounted weapons need it and you'd know it if you used them. The concern is a line of aim mode snipers, not a single machinegun.

**Comment by SimplySmiley123** [comment]
> Stock snipers like the SR-127 are pretty niche and i don't see as often, and are substantially affected by their low fire-rate since they need to aim 90% of the time. The only thing getting killed by two snipers is like.... a stunlocked runner. They merely exist to deal damage through units and let chasers finish the job.
Ive seen different changes for these...I dare say that limiting the amount of them on vendors would be inconsequential. But nerfing its range to basically a mounted gun can seriously kill these guns.

The only saving grace of those was that they had the long range. They don't deal nearly enough damage to be a threat unless you stack 4-5 marines with them and they somehow are coordinated enough to always aim at the same xeno at once and fire at the same time.
They're abysmall dogshit at miniscope to no scope range and at that point, you're genuinely far better off grabbing a bog-standard AR instead. 
If you want those to be miniscope range weapons and restricted ones at that, you would need to rework them to not be useless at that range.

> And now for the mounted/heavy ones like AT36, AGLs... These are weapons that require cover/support and good positioning to be used effectively, and can be easily shut down if flankers exist/marine is distracted. Plus, you can break the guns or just stun the gunner with any CC ability, and they've got to be insanely careful with their friendly fire.
By nerfing their scopes into less than a miniscope.... you are turning these guns into abysmal dogshit.

Yup. Those will just become FF griefing machines with miniscope / no scope.
Especially AGL. That thing even with a long range scope is easy to FF with, without it it's a nigh guarantee to FF marines if you want to set it up anywhere that isn't right in front of a xeno's face.

**Comment by Woker2004** [comment]
Also AGLS NEEDS RANGE to work, you take a good 10 seconds to setup and pickup, and truthfully the tides change VERY QUICKLY, a miniscope ranged AGLS would be pure grieving tool

**Comment by Woker2004** [comment]
Also majority of the time if your using a scope, YOUR IN AIM MODE, your literally taking the aim mode Reduced firerate regardless, there is a reason why many AR's don't use scopes, snipers use their scopes to see and attempt to suppress fire, Boiler can shoot straight through a xenos, a marine can not shoot straight through a marine unless on aim mode

**Comment by Wisemonster** [comment]
Alright, thanks to Lumi and Xander, I've finally got pop scaling to work.

**Comment by SiegeB0t** [comment]
### Miniscopes are cringe.
It is unfortunate that we are resorting to blanket mechanic removals instead of buffing mechanics that are explicitly designed to counter marine range (fog of war, resin, xeno armor, xeno speed, gas, flare removal, stuns, arm fracs, lower accuracy at range, bullet damage loss at range). 

Marine range is only strong in the case where xenos don't have resin and the area is flared. Removing marine range means a hurt marine force has little ability to skirmish when falling back. The removal only promotes more degenerate play styles (less xeno resin in favor of xeno unga, using inaccurate walls of bullets as a shield).

I would much rather see a removal of the offending guns rather than a removal of an entire mechanic. None of these "op" guns saw much use in the first place.

Another possible fix is buffing resin walls to bullets at range.

Another fix is to make aim mode fail softly; 50% ff chance at a distance would prevent stacking.

Another fix is to make flares weaker to acid.

**Comment by Kuro020** [comment]
Like i said two days ago, after the xeno buff i dont feel this pr is necessary anymore as xenos dont instadie to these things now, amr will be having its own public execution pr

**Pull request closed without merging**